stable/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -157,16 +157,30 @@ spec:
             path: /healthz
             scheme: HTTPS
 {{- end }}
-{{- if .Values.clusterAgent.confd }}
         volumeMounts:
+{{- if .Values.clusterAgent.confd }}        
           - name: confd
             mountPath: /conf.d
             readOnly: true
+{{- end}}
+{{- if .Values.clusterAgent.datadog_cluster_yaml }}
+          - name: cluster-agent-yaml
+            mountPath: /etc/datadog-agent/datadog-cluster.yaml
+            subPath: datadog-cluster.yaml
+            readOnly: true
+{{- end}}
       volumes:
+{{- if .Values.clusterAgent.confd }}
```

Comment:
If users are relying volume without using the option, it will not be mounted anymore I believe.

Reply:
The confd mount was the only mount here before, and it wrapped the whole volumes and volumeMounts sections.
I had to change it so that multiple mounts could be added.